sourced/ml/transformers/indexer.py

```diff
@@ -53,21 +56,30 @@ def values(self):
         return arr
 
     def calculate_value_to_index(self, rdd: RDD):
-        column_name = self.column
-        if isinstance(column_name, str):
-            column = rdd.map(lambda x: getattr(x, column_name))
+        if self.pickle_index and os.path.exists(self.pickle_index):
+            self._log.info("Unpickling the index from %s", self.pickle_index)
+            with open(self.pickle_index, 'rb') as _in:
+                self._value_to_index = pickle.load(_in)
         else:
-            column = rdd.map(lambda x: x[column_name])
-        self._log.info("Collecting the list of distinct sorted values (%s)", column_name)
-        values = column.distinct()
-        if self.explained:
-            self._log.info("toDebugString():\n%s", values.toDebugString().decode())
-        values = values.collect()
-        values.sort()  # We do not expect an extraordinary number of distinct values
-        self._log.info("%d distinct values", len(values))
-        if len(values) == 0:
-            raise RuntimeError("Number of distinct values is zero.")
-        self._value_to_index = {d: i for i, d in enumerate(values)}
+            column_name = self.column
+            if isinstance(column_name, str):
+                column = rdd.map(lambda x: getattr(x, column_name))
+            else:
+                column = rdd.map(lambda x: x[column_name])
+            self._log.info("Collecting the list of distinct sorted values (%s)", column_name)
+            values = column.distinct()
+            if self.explained:
+                self._log.info("toDebugString():\n%s", values.toDebugString().decode())
+            values = values.collect()
+            values.sort()  # We do not expect an extraordinary number of distinct values
+            self._log.info("%d distinct values", len(values))
+            if len(values) == 0:
+                raise RuntimeError("Number of distinct values is zero.")
+            self._value_to_index = {d: i for i, d in enumerate(values)}
```

Comment:
The data scheme is the same as in DocumentFrequencies model which is faster and more efficient than pickling. I suggest to use it instead.

```python
DocumentFrequencies().construct(self._value_to_index).save("...")
```